File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;
class Solution {
    private static int answer = 0;
    
    private static void dfs(int[] numbers, int target, int sum, int depth){
        if (depth == numbers.length){
            if (target == sum) answer++;
        } else{
            dfs(numbers, target, sum+numbers[depth], depth+1);
            dfs(numbers, target, sum-numbers[depth], depth+1);
        }
    }
    
    
    public int solution(int[] numbers, int target) {
        dfs(numbers, target, 0, 0);
        return answer;
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(numbers, target, current_sum, depth):
    global answer
    if depth == len(numbers):
        if current_sum == target:
            answer += 1
    else:
        dfs(numbers, target, current_sum + numbers[depth], depth + 1)
        dfs(numbers, target, current_sum - numbers[depth], depth + 1)

def solution(numbers, target):
    global answer
    answer = 0
    dfs(numbers, target, 0, 0)
    return answer
